src/codecs/png.rs

```diff
@@ -532,6 +532,9 @@ impl<W: Write> PngEncoder<W> {
     }
 
     /// Encodes the image `data` that has dimensions `width` and `height` and `ColorType` `c`.
+    ///
+    /// Expects data in big endian.
+    #[deprecated = "Use `PngEncoder::write_image` instead. Beware that `write_image` has a different endianness convention"]
     pub fn encode(self, data: &[u8], width: u32, height: u32, color: ColorType) -> ImageResult<()> {
```

Comment:
Maybe add `#[doc(hidden)]`?

Reply:
Hm, if we do that then users won't be able to see that the function expected big endian. (Older releases didn't mention it)

Your call

Reply:
I guess we can have it deprecated for a couple of versions before hiding it :+1: